### PR TITLE
Bazel support for //test/common/{common,event}/... (#415).

### PR DESCRIPTION
**Pull request by htuch**
In addition to the handful of additional tests, this patch introduces additional rules covering ~30%
of the build targets we will need for Envoy build/test, which are transitiviely depended on by the
tests. External deps added for boringssl, c-ares, libevent and tclap.

### CODE REVIEW

**Comment by htuch** [comment]
@lizan @PiotrSikora for review.

**Comment by lizan** [comment]
Can you update this PR align with #602 too? Thanks.

**Comment by htuch** [comment]
@lizan @PiotrSikora This is now rebased on #602 and other recent changes. PTAL.

File: source/common/event/dispatcher_impl.cc [hunk 1]
```diff
@@ -1,11 +1,10 @@
-#include "dispatcher_impl.h"
-#include "file_event_impl.h"
-#include "signal_impl.h"
-#include "timer_impl.h"
-
 #include "envoy/network/listener.h"
 #include "envoy/network/listen_socket.h"
 
+#include "common/event/dispatcher_impl.h"
```
Comment (lizan) [inline source/common/event/dispatcher_impl.cc]: Keep related header on the top? (throughout)
https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

**Pull request merged**